### ISSUE TITLE
ukt necessity

### ISSUE THREAD

**Issue by anders34** [uwhpsc-2016/homework4#10]
I'm a little confused about in heat.c, why we have ukt. If we already have uk, why do we need a duplicate of it at ukt.

**Comment by anders34** [comment]
Because uk is a private variable to each thread/process, right? So we shouldn't need to make ukt...

**Comment by swcarmer** [comment]
I agree, we could use the `uk` pointer passed in as our `ukt`, which negates the need for the array copies as well.

**Comment by quantheory** [comment]
It's true that `ukt` is unnecessary in this case. If there were two separate "input" and "output" pointers, and you weren't allowed to change the "input" array, then it would be necessary to make a copy. But in this case, it's admittedly an extra step.
